File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    const answer =  numbers.sort((a, b) => {
        a = String(a)
        b = String(b)
        
        if (a + b < b + a) {
            return 1;
        } else if (a + b == b + a) {
            return 0;
        } else {
            return -1;
        }
    }).join('')
    
    return answer[0] === '0' ? '0' : answer
}

File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def comparator(a, b):
    a, b = str(a), str(b)
    c, d = int(a + b), int(b + a)
    if c > d:
        return -1
    if c == d:
        return 0
    return 1

def solution(numbers):
    return str(int("".join(list(map(str, sorted(numbers, key=cmp_to_key(comparator)))))))